server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -219,6 +220,7 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
             new ScalingExecutorBuilder(Names.FETCH_SHARD_STARTED, 1, 2 * allocatedProcessors, TimeValue.timeValueMinutes(5), false)
         );
         builders.put(Names.FORCE_MERGE, new FixedExecutorBuilder(settings, Names.FORCE_MERGE, 1, -1, false));
+        builders.put(Names.COORDINATION, new ScalingExecutorBuilder(Names.COORDINATION, 1, 1, TimeValue.timeValueSeconds(30), false));
```

Comment:
It would be nice to not introduce a new scaling thread pool that does'nt handle rejections after shutdown. Given the limited area where this thread pool is used I think we could handle rejection everywhere and set the `rejectAftershutdown` flag to `true` here.

Reply:
++ I made it a fixed pool of size 1 now. And fixed the one spot where this could have failed that I found.